src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -181,39 +168,29 @@ public PriorityScheduler(int corePoolSize, int maxPoolSize,
   public PriorityScheduler(int corePoolSize, int maxPoolSize,
                            long keepAliveTimeInMs, TaskPriority defaultPriority, 
                            long maxWaitForLowPriorityInMs, ThreadFactory threadFactory) {
-    ArgumentVerifier.assertGreaterThanZero(corePoolSize, "corePoolSize");
-    if (maxPoolSize < corePoolSize) {
-      throw new IllegalArgumentException("maxPoolSize must be >= corePoolSize");
-    }
-    
-    //calls to verify and set values
-    setKeepAliveTime(keepAliveTimeInMs);
-    setMaxWaitForLowPriority(maxWaitForLowPriorityInMs);
-    
+    this(new WorkerPool(threadFactory, corePoolSize, maxPoolSize, 
```

Comment:
Hum, I feel like there is a better model here.  I almost feel like WorkerPool and StatisticWorkerPool should be in there own files.  This feels like a case where multiple inheritance could solve this well.